static/js/user/controllers/user.controller.js

```diff
@@ -10,16 +10,50 @@
         .controller('UserController', UserController);
 
     UserController.$inject = ['$location', '$scope',
-        '$window', '$mdToast', '$mdDialog', 'Authentication', 'User', 'Payment'];
+        '$window', '$mdToast', '$mdDialog', 'Authentication', 'User', 'Payment', '$http'];
 
     /**
      * @namespace UserController
      */
     function UserController($location, $scope,
-                            $window, $mdToast, $mdDialog, Authentication, User, Payment) {
+                            $window, $mdToast, $mdDialog, Authentication, User, Payment, $http) {
 
         var vm = this;
         vm.paypal_payment = paypal_payment;
+        vm.toggleEdit = toggleEdit;
+        vm.updateProfileDesc = updateProfileDesc;
+        vm.edit=false;
+        vm.genders=[
+            {"key":"M", "value":"Male"},
+            {"key":"F", "value":"Female"},
+            {"key":"O", "value":"Other"}
+        ];
+        vm.countries = [];
+        vm.cities = [];
+
+        $http({
```

Comment:
we interact with the api using services, that way we can re-use the methods anywhere, would be better to move this call to there and call it accordingly from the controller